doc/source/whatsnew/v0.25.1.rst

```diff
@@ -85,6 +85,7 @@ Interval
 Indexing
 ^^^^^^^^
 
+- Break reference cycle in :meth:`Index._engine` (:issue:`27585`)
```

Comment:
Index._engine isn't part of the API docs. Phrase as "Break reference cycle involving :class:\`Index\`."